test/integration/targets/win_psmodule/tasks/tests.yml

```diff
@@ -0,0 +1,810 @@
+# This file is part of Ansible
+
+# test code for the win_psmodule module when using winrm connection
+# Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
+# Copyright: (c) 2017, Daniele Lazzari <lazzari@mailup.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+---
+
+- name: "get the latest version of the {{ powershell_module }} module in PowerShell Gallery"
+  win_shell: "((Find-Module -Name {{ powershell_module }}).Version).ToString()"
+  changed_when: false
+  register: module_latest_version
+
+- name: "get {{ skip_publisher_check_module }} is preinstalled"
+  win_shell: "(Get-Module -Name {{ skip_publisher_check_module }} -ListAvailable -ErrorAction Ignore | Measure-Object).Count"
+  register: skip_publisher_module_exist
+
+- name: "check installing module from Powershell Gallery - without version - check mode: true"
+  win_psmodule:
+    name: "{{ powershell_module }}"
+    state: present
+  check_mode: yes
+  register: module_without_version_setup_1
+
+- name: get result installing module from Powershell Gallery - without version
+  win_shell: "(Get-Module -Name {{ powershell_module }} -ListAvailable -ErrorAction Ignore | Where-Object { $_.Version -eq '{{ module_latest_version.stdout | trim }}' } | Measure-Object).Count -eq 1"
+  changed_when: false
+  register: result_module_without_version_setup_1
+
+- name: "test installing module from Powershell Gallery - without version - check mode: true"
+  assert:
+    that:
+      - "module_without_version_setup_1 is changed"
+      - "{{ result_module_without_version_setup_1.stdout | trim }} | bool == false"
```

Comment:
You don't need to use jinja2 block for assert, this can simply be `result_module_without_version_setup_1.stdout | trim | bool == False`